Core/src/org/sleuthkit/autopsy/casemodule/AddImageTask.java

```diff
@@ -73,40 +76,26 @@
-     *                             (e.g., a UUID).
-     * @param imagePath            Path to the image file.
-     * @param sectorSize           The sector size (use '0' for autodetect).
-     * @param timeZone             The time zone to use when processing dates
-     *                             and times for the image, obtained from
-     *                             java.util.TimeZone.getID.
-     * @param ignoreFatOrphanFiles Whether to parse orphans if the image has a
-     *                             FAT filesystem.
-     * @param md5                  The MD5 hash of the image, may be null.
-     * @param sha1                 The SHA-1 hash of the image, may be null.
-     * @param sha256               The SHA-256 hash of the image, may be null.
-     * @param imageWriterPath      Path that a copy of the image should be
-     *                             written to. Use empty string to disable image
-     *                             writing
+     * 
+     * @param imageDetails         Holds all data about the image.
      * @param progressMonitor      Progress monitor to report progress during
      *                             processing.
-     * @param callback             Callback to call when processing is done.
+     * @param addDataSourceCallbacks  Callback for sending data to the ingest pipeline if an ingest stream is being used.
+     * @param addImageTaskCallback    Callback for dealing with add image task completion.
      */
-    AddImageTask(String deviceId, String imagePath, int sectorSize, String timeZone, boolean ignoreFatOrphanFiles, String md5, String sha1, String sha256, ImageWriterSettings imageWriterSettings,
-            DataSourceProcessorProgressMonitor progressMonitor, DataSourceProcessorCallback callback) {
-        this.deviceId = deviceId;
-        this.imagePath = imagePath;
-        this.sectorSize = sectorSize;
-        this.timeZone = timeZone;
-        this.ignoreFatOrphanFiles = ignoreFatOrphanFiles;
-        this.md5 = md5;
-        this.sha1 = sha1;
-        this.sha256 = sha256;
-        this.imageWriterSettings = imageWriterSettings;
-        this.callback = callback;
+    AddImageTask(ImageDetails imageDetails, DataSourceProcessorProgressMonitor progressMonitor, AddDataSourceCallbacks addDataSourceCallbacks,  
+            AddImageTaskCallback addImageTaskCallback) {
+        this.deviceId = imageDetails.deviceId;
+        this.imagePath = imageDetails.imagePath;
+        this.sectorSize = imageDetails.sectorSize;
+        this.timeZone = imageDetails.timeZone;
+        this.ignoreFatOrphanFiles = imageDetails.ignoreFatOrphanFiles;
+        this.md5 = imageDetails.md5;
+        this.sha1 = imageDetails.sha1;
+        this.sha256 = imageDetails.sha256;
+        this.imageWriterSettings = imageDetails.imageWriterSettings;
+        this.addDataSourceCallbacks = addDataSourceCallbacks;
+        this.addImageTaskCallback = addImageTaskCallback;
```

Comment:
Perhaps just have an ImageDetails instance variable instead of duplicating the ImageDetails class within the AddImageTask class?